plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/HiveMetastoreTableOperations.java

```diff
@@ -88,21 +88,15 @@ protected void commitToExistingTable(TableMetadata base, TableMetadata metadata)
                         .setParameter(METADATA_LOCATION_PROP, newMetadataLocation)
                         .setParameter(PREVIOUS_METADATA_LOCATION_PROP, currentMetadataLocation)
                         .build();
+
+                // todo privileges should not be replaced for an alter
+                PrincipalPrivileges privileges = table.getOwner().map(MetastoreUtil::buildInitialPrivilegeSet).orElse(NO_PRIVILEGES);
+                metastore.replaceTable(database, tableName, table, privileges);
             }
             catch (RuntimeException e) {
-                try {
```

Comment:
As per https://github.com/apache/iceberg/issues/2317

> basically we need to group failures into two categories:

> 1. Failures that are reports from the Catalog that it could not perform the operation for some reason
> 2. Failures where the client has lost contact for some reason (basically everything else)

This implementation never deletes the new metadata file. Is this intentional?

See https://github.com/apache/iceberg/pull/2328/files#diff-e502621d52f86cf0ec3187dda30ac61f6b76efb7b6276bc8d233ccb2c836fb98R482 where the metadata is deleted only in case of `FAILURE`

Reply:
It is intentional, we do have a little different logic than iceberg library so it can't be copied one to one, also it is safer not to ever delete these files.